src/metrics/loc.rs

```diff
@@ -500,9 +500,59 @@ impl Loc for CppCode {
     }
 }
 
+impl fmt::Display for Java {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+impl Loc for JavaCode {
+    fn compute(node: &Node, stats: &mut Stats, is_func_space: bool, is_unit: bool) {
+        use Java::*;
+
+        let (start, end) = init(node, stats, is_func_space, is_unit);
+        let kind_id: Java = node.object().kind_id().into();
+        match kind_id {
+            Program => {}
+            Comment => {
+                add_cloc_lines(stats, start, end);
+            }
+            AssertStatement
```

Comment:
Maybe you missed `InstanceofExpression` and few others, have a look on:
https://github.com/tree-sitter/tree-sitter-java/blob/a24ae7d16de3517bff243a87d087d0b4877a65c5/grammar.js#L161-L170

Reply:
@dburriss ^^

Reply:
@calixteman  I should have mentioned it explicitly. I did add these in but it completely breaks the counts. I will record some examples. Maybe we can get to the same counts with fewer statements and more expressions.

Reply:
For ease of discussion, here is the result of parsing that:
```
LocalVariableDeclaration
TypeIdentifier
VariableDeclarator
Identifier
EQ
TernaryExpression
BinaryExpression
ParenthesizedExpression
LPAREN
BinaryExpression
Identifier
PERCENT
DecimalIntegerLiteral
RPAREN
EQEQ
DecimalIntegerLiteral
QMARK
StringLiteral
COLON
StringLiteral
SEMI
Comment
```

Reply:
I re-read the sloc & lloc definitions on wikipedia:
https://en.wikipedia.org/wiki/Source_lines_of_code#Measurement_methods
and in lloc we should have only statements.
I wonder why I talked about InstanceofExpression... maybe because there was an Expression here in the 1st version of your patch.... or maybe I was a bit tired... Anyway sorry for that.
For example with js we only have statements for logical_lines:
https://github.com/mozilla/rust-code-analysis/blob/d488b6eb5b4de522700a75a84602d404a5f64de1/src/metrics/loc.rs#L343-L349
But for rust code:
https://github.com/mozilla/rust-code-analysis/blob/d488b6eb5b4de522700a75a84602d404a5f64de1/src/metrics/loc.rs#L424-L436
the term `Expression` is used in place of `Statement`: it's a bit confusing.
So it really depends on the way the grammar is built itself.

Anyway, for Java we've a a grammar definition for statement:
https://github.com/tree-sitter/tree-sitter-java/blob/a24ae7d16de3517bff243a87d087d0b4877a65c5/grammar.js#L415
so I guess we can basically re-use what we've here and nothing else (like for js, python stuff).

Reply:
Yeah definitely depends on the language. I guess Rust takes some FP cues so expressions are preferred. Statements were the original push as discussed with Luni-4 https://github.com/mozilla/rust-code-analysis/issues/359#issuecomment-952631456. Sorrry, I should have thought to link to it sooner.
I'll revert back in the next day or 2. I learned a bit about Java. Have not touched it since version 6 so if nothing else I learned a few things about how it does lambdas. I must admit, I like it even less than I like its generics but hey. Each to their own.

Reply:
All expressions are removed now.